clang/test/SemaSYCL/intel-reqd-work-group-size-device-direct-prop.cpp

```diff
@@ -0,0 +1,146 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -fsyntax-only -verify -DTRIGGER_ERROR %s
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -ast-dump %s | FileCheck %s
+
+// Test for AST of reqd_work_group_size kernel attribute in SYCL 2020.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+#ifndef __SYCL_DEVICE_ONLY__
+// expected-no-diagnostics
+class Functor {
+public:
+  [[sycl::reqd_work_group_size(4)]] void operator()() const {}
+};
+
+void bar() {
+  q.submit([&](handler &h) {
+    Functor f;
+    h.single_task<class kernel_name>(f);
+  });
+}
+
+#else
+#ifdef TRIGGER_ERROR
+class Functor32 {
+public:
+  [[cl::reqd_work_group_size(32)]] void operator()() const {} // expected-error {{'reqd_work_group_size' attribute requires exactly 3 arguments}} \
```

Comment:
Also, is there a reason why you moved these diagnostics over to this patch. It is not really related to this change and in my opinion, making this patch larger and more confusing. I personally think we should just leave the diagnostics as it was and then refactor tests in a follow-up patch if we want the diagnostics for SYCL2020. I'm not convinced we even need this refactoring since it really hasn't changed between versions

Reply:
>>Also, is there a reason why you moved these diagnostics over to this test.

Yes, the deprecation warnings happens with SYCL 2020 mode. If we keep same tests, it will very messy and there is no way we can pass ``FILECHECK`` to verify the changes for sycl 2017 vs SYCL 2020 modes. Previously we did not pass any std option to the RUN line so the test was taking -sycl-std=2020 by default since default version is SYCL 2020.

``// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -Wno-sycl-2017-compat -fsyntax-only -verify -DTRIGGER_ERROR %s
// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -Wno-sycl-2017-compat -ast-dump %s | FileCheck %s
``

Reply:
Ok. Thanks for explaining. Even if you don't do it in this patch, we need to start splitting diagnostic tests and AST tests. The tests are getting hard to follow because there is so much happening

Reply:
> Ok. Thanks for explaining. Even if you don't do it in this patch, we need to start splitting diagnostic tests and AST tests. The tests are getting hard to follow because there is so much happening

@elizabethandrews, i agree, I will do this in a separate patch. Tahnks